web-components/src/components/table-advanced/TableAdvanced.ts

```diff
@@ -157,64 +179,63 @@ export namespace TableAdvanced {
 
     private validateData() {
       const lenNodes = this.COLS.length;
-       // validate
-
-       const lenData = this.ROWS.reduce((acc, d) => acc + d.length, 0);
-       if (lenData % lenNodes != 0) {
-         this.error = this.error =
-           "DATA ERROR: Data length mismatch. You must provide (numberOfRows * numberOfColumns) amount of data values.";
-         return;
-       }
- 
-       this.ROWS.forEach((d, i) => {
-         const len = d.length;
-         if (len != lenNodes) {
-           this.error = `DATA ERROR: Total number of cols (=${lenNodes}) and data[${i}] length (=${len}) mismatch`;
-         }
-       });
+      // validate
+
+      const lenData = this.ROWS.reduce((acc, d) => acc + d.length, 0);
+      if (lenData % lenNodes != 0) {
+        this.error = this.error =
+          "DATA ERROR: Data length mismatch. You must provide (numberOfRows * numberOfColumns) amount of data values.";
+        return;
+      }
+
+      this.ROWS.forEach((d, i) => {
+        const len = d.length;
+        if (len != lenNodes) {
+          this.error = `DATA ERROR: Total number of cols (=${lenNodes}) and data[${i}] length (=${len}) mismatch`;
+        }
+      });
     }
 
     private populateTemplate() {
-       // TEMPLATES
-       const templates = this.config.cellTemplates;
-       const templatesKeys = Object.keys(templates || {});
-       if (templates && templatesKeys.length) {
-         this.ROWS.forEach((row, iRow) => {
-           row.forEach((cell, iCol) => {
-             for (const k in templates) {
-               const idx = cell.text.indexOf(k);
-               if (idx != -1) {
-                 const t = templates[k];
-                 const template = this.querySelector<HTMLTemplateElement>(`#${t.templateName}`);
-                 if (template == null) {
-                   console.warn(`cellTemplates["${k}"]: Missing '${t.templateName}' template.`);
-                   continue;
-                 }
- 
-                 let text = cell.text.replace(k, "");
-                 if (t.contentCb) {
-                   text = t.contentCb({ col: iCol, row: iRow, content: text, insertIndex: idx });
-                 }
- 
-                 this.ROWS[iRow][iCol] = {
-                   text,
-                   template: {
-                     template,
-                     templateCb: t.templateCb,
-                     insertIndex: t.contentUse == "replace" ? -1 : idx
-                   }
-                 };
-                 break;
-               }
-             }
-           });
-         });
-       }
- 
+      // TEMPLATES
+      const templates = this.tableConfig.cellTemplates;
+      const templatesKeys = Object.keys(templates || {});
+      if (templates && templatesKeys.length) {
+        this.ROWS.forEach((row, iRow) => {
+          row.forEach((cell, iCol) => {
```

Comment:
can you refactor? what is happening here its not very readable

Reply:
Not made any change here. Although here the templates are being populated